python/spglib/spglib.py

```diff
@@ -97,6 +103,101 @@ class SpglibError:
 spglib_error = SpglibError()
 
 
+@dataclasses.dataclass(eq=True, frozen=True)
+class SpaceGroupType(Mapping[str, Any]):
+    """Space group type information.
+
+    Dictionary keys are as follows:
+
+    - number : int
+        International space group number
+    - international_short : str
+        International short symbol.
+    - international_full : str
+        International full symbol.
+    - international : str
+        International symbol.
+    - schoenflies : str
+        Schoenflies symbol.
+    - hall_number : int
+        Hall symbol ID number.
+    - hall_symbol : str
+        Hall symbol.
+    - choice : str
+        Centring, origin, basis vector setting.
+    - pointgroup_international : str
+        International symbol of crystallographic point group.
+    - pointgroup_schoenflies : str
+        Schoenflies symbol of crystallographic point group.
+    - arithmetic_crystal_class_number : int
+        Arithmetic crystal class number
+    - arithmetic_crystal_class_symbol : str
+        Arithmetic crystal class symbol.
+    """
+
+    number: int
+    international_short: str
+    international_full: str
+    international: str
+    schoenflies: str
+    hall_number: int
+    hall_symbol: str
+    choice: str
+    pointgroup_international: str
+    pointgroup_schoenflies: str
+    arithmetic_crystal_class_number: int
+    arithmetic_crystal_class_symbol: str
+
+    def __getitem__(self, key: str) -> Any:
+        """Return the value of the key."""
```

Comment:
Isn't there a built-in docstring for these that is inherited from abstract? So far I [guess not](https://github.com/python/cpython/blob/656491783d2be0fe36feef5085a2406a05abbd66/Lib/_collections_abc.py#L800-L802) :shrug:

Reply:
I don't think there is.